src/client/render/core.cpp

```diff
@@ -19,104 +19,55 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 */
 
 #include "core.h"
-#include "client/camera.h"
-#include "client/client.h"
-#include "client/clientmap.h"
-#include "client/hud.h"
-#include "client/minimap.h"
+#include "plain.h"
 #include "client/shadows/dynamicshadowsrender.h"
+#include "settings.h"
+
 
 RenderingCore::RenderingCore(IrrlichtDevice *_device, Client *_client, Hud *_hud)
-	: device(_device), driver(device->getVideoDriver()), smgr(device->getSceneManager()),
-	guienv(device->getGUIEnvironment()), client(_client), camera(client->getCamera()),
-	mapper(client->getMinimap()), hud(_hud),
-	shadow_renderer(nullptr)
+	: device(_device), client(_client), hud(_hud), shadow_renderer(createShadowRenderer(device, client)), pipeline(new RenderPipeline()), 
+	virtual_size_scale(1.0f)
 {
-	screensize = driver->getScreenSize();
-	virtual_size = screensize;
+}
 
-	if (g_settings->getBool("enable_shaders") &&
-			g_settings->getBool("enable_dynamic_shadows")) {
-		shadow_renderer = new ShadowRenderer(device, client);
-	}
+RenderingCore::RenderingCore(IrrlichtDevice *_device, Client *_client, Hud *_hud, 
```

Comment:
Why do we need two different constructors? When are we using which one?

Reply:
One of them was a legacy, removed now.